### PR TITLE
Allow messages to be sent using the non-persistent delivery mode

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

File: src/NServiceBus.Transport.RabbitMQ.Tests/When_sending_a_message_over_rabbitmq.cs [hunk 1]
```diff
@@ -72,17 +72,27 @@ public Task Should_set_correlation_id_if_present()
...
+                Assert.False(message.Headers.ContainsKey(BasicPropertiesExtensions.UseNonPersistentDeliveryHeader), "Temp header should be removed before sending the message");
+            });
         }
 
         [Test]
-        public Task Should_preserve_the_recoverable_setting_if_set_to_non_durable()
+        public Task Should_preserve_the_recoverable_setting_if_set_to_durable()
         {
-            return Verify(new OutgoingMessageBuilder().NonDurable(), result => Assert.True(result.Headers[Headers.NonDurableMessage] == "True"));
+            return Verify(new OutgoingMessageBuilder(), result => Assert.True(result.Headers[Headers.NonDurableMessage] == "False"));
```
Comment (ramonsmits) [inline src/NServiceBus.Transport.RabbitMQ.Tests/When_sending_a_message_over_rabbitmq.cs:87]:
```suggestion
            return Verify(new OutgoingMessageBuilder(), result => Assert.True(result.Headers[Headers.NonDurableMessage] == bool.FalseString));
```

File: src/NServiceBus.Transport.RabbitMQ/Configuration/RabbitMQTransportOptionsExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="options">Options being extended.</param>
+        public static void UseNonPersistentDeliveryMode(this SendOptions options)
+        {
+            Guard.AgainstNull(nameof(options), options);
+
+            options.SetHeader(BasicPropertiesExtensions.UseNonPersistentDeliveryHeader,true.ToString());
```
Comment (ramonsmits) [inline src/NServiceBus.Transport.RabbitMQ/Configuration/RabbitMQTransportOptionsExtensions.cs:18]:
```suggestion
            options.SetHeader(BasicPropertiesExtensions.UseNonPersistentDeliveryHeader, bool.TrueString);
```

File: src/NServiceBus.Transport.RabbitMQ/Configuration/RabbitMQTransportOptionsExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="options">Options being extended.</param>
+        public static void UseNonPersistentDeliveryMode(this PublishOptions options)
+        {
+            Guard.AgainstNull(nameof(options), options);
+
+            options.SetHeader(BasicPropertiesExtensions.UseNonPersistentDeliveryHeader, true.ToString());
```
Comment (ramonsmits) [inline src/NServiceBus.Transport.RabbitMQ/Configuration/RabbitMQTransportOptionsExtensions.cs:29]:
```suggestion
            options.SetHeader(BasicPropertiesExtensions.UseNonPersistentDeliveryHeader, bool.TrueString);
```

File: src/NServiceBus.Transport.RabbitMQ/Configuration/RabbitMQTransportOptionsExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="options">Options being extended.</param>
+        public static void UseNonPersistentDeliveryMode(this ReplyOptions options)
+        {
+            Guard.AgainstNull(nameof(options), options);
+
+            options.SetHeader(BasicPropertiesExtensions.UseNonPersistentDeliveryHeader, true.ToString());
```
Comment (ramonsmits) [inline src/NServiceBus.Transport.RabbitMQ/Configuration/RabbitMQTransportOptionsExtensions.cs:40]:
```suggestion
            options.SetHeader(BasicPropertiesExtensions.UseNonPersistentDeliveryHeader, bool.TrueString);
```

**Review by ramonsmits — CHANGES_REQUESTED** [review]
Removal of ToStrings and string literals.

**Comment by andreasohlund** [comment]
I will wrap this up once https://github.com/Particular/NServiceBus/pull/5785 is merged and we have a core alpha to compile against

**Review by lailabougria — APPROVED** [review]

File: src/NServiceBus.Transport.RabbitMQ.Tests/When_sending_a_message_over_rabbitmq.cs [hunk 5]
```diff
@@ -72,15 +72,13 @@ public Task Should_set_correlation_id_if_present()
...
-        }
-
-        [Test]
-        public Task Should_preserve_the_recoverable_setting_if_set_to_non_durable()
-        {
-            return Verify(new OutgoingMessageBuilder().NonDurable(), result => Assert.True(result.Headers[Headers.NonDurableMessage] == "True"));
+            return Verify(new OutgoingMessageBuilder().WithHeader(BasicPropertiesExtensions.UseNonPersistentDeliveryHeader, true.ToString()), (message, basicDeliverEventArgs) =>
+            {
+                Assert.False(basicDeliverEventArgs.BasicProperties.Persistent);
+                Assert.True(message.Headers.ContainsKey(BasicPropertiesExtensions.UseNonPersistentDeliveryHeader), "Temp header should not removed to make sure that retries keeps the setting");
```
[the remark this answers is not in the document]
Comment (andreasohlund) [reply src/NServiceBus.Transport.RabbitMQ.Tests/When_sending_a_message_over_rabbitmq.cs:80]: The reasoning is that by keeping the header retries will not loose this setting, eg doing retries via SC. Does that make sense?

**Pull request merged**